diff --git a/node_modules/react-native-dialogflow/index.js b/node_modules/react-native-dialogflow/index.js
index 8b9e018..4e724ce 100644
--- a/node_modules/react-native-dialogflow/index.js
+++ b/node_modules/react-native-dialogflow/index.js
@@ -53,6 +53,7 @@ dialogflow2.setConfiguration = async function (clientEmail, privateKey, language
     dialogflow2.sessionId = dialogflow2.sessionId ? dialogflow2.sessionId : dialogflow2.guid();
 
     // set listeners
+
     Voice.onSpeechStart = (c) => {
         dialogflow2.speechResult = null;
         if (dialogflow2.onListeningStarted) {
@@ -78,16 +79,20 @@ dialogflow2.setConfiguration = async function (clientEmail, privateKey, language
     }
 
     Voice.onSpeechResults = (result) => {
+        clearTimeout(dialogflow2.silenceTimer)
         if (result.value) {
             dialogflow2.speechResult = result.value;
         }
+        dialogflow2.silenceTimer = setTimeout(() => {
+            Voice.stop()
+        }, 1500)
+
     }
 }
 
 dialogflow2.startListening = function (onResult, onError) {
     dialogflow2.onResult = onResult;
     dialogflow2.onError = onError;
-
     Voice.start(dialogflow2.languageTag);
 }
 
@@ -96,4 +101,3 @@ dialogflow2.finishListening = function () {
 }
 
 export { dialogflow2 as Dialogflow_V2 };
-
